core/arch/arm/include/mm/core_memprot.h

```diff
@@ -45,14 +45,13 @@
 
 /* memory atttributes */
 enum buf_is_attr {
-	CORE_MEM_SEC,
+	CORE_MEM_CACHED,
+	CORE_MEM_EXTRAM,
+	CORE_MEM_NSEC_SHM,
 	CORE_MEM_NON_SEC,
+	CORE_MEM_SEC,
 	CORE_MEM_TEE_RAM,
 	CORE_MEM_TA_RAM,
-	CORE_MEM_NSEC_SHM,
-	CORE_MEM_EXTRAM,
-	CORE_MEM_INTRAM,
-	CORE_MEM_CACHED,
```

Comment:
Indeed this PR is NOT a fix and this does not aim at fixing bad implementations. Type verifications a compile time could help, but i am not sure `enum` types are good at this.

I though about your solution but won't this decrease optims of the switch/case implementations, without providing a real bullet proof solution ?

I feel a good solution would be to fully review the memory area and attributes definition and handling. But i have nothing to propose...

Jens, your opinion ?